doc/en/conf.py

```diff
@@ -112,6 +112,16 @@
 # A list of ignored prefixes for module index sorting.
 # modindex_common_prefix = []
 
+# A list of links to be ignored by linkcheck builder
+linkcheck_ignore = [
+    "https://github.com/numpy/numpy/blob/master/doc/release/1.16.0-notes.rst#new-deprecations",
+    "https://blogs.msdn.microsoft.com/bharry/2017/06/28/testing-in-a-cloud-delivery-cadence/",
+    "http://pythontesting.net/framework/pytest-introduction/",
+]
```

Comment:
this link can be used to maintain false-positives

Reply:
Can those be regular expressions?
(Might be worth adding a link to its doc there - and sorry for being too lazy to search for it myself)

Reply:
They can be regular expressions. I updated the comment accordingly. 
Thanks for the suggestion.

Reply:
Thanks.

Why is "https://github.com/numpy/numpy/blob/master/doc/release/1.16.0-notes.rst#new-deprecations" ignored?

I also suggest to ignore `https://github.com/pytest-dev/pytest/issues/\d+` (a lot of links, and those should be OK in general)

Reply:
Also:

- linkcheck_workers=20 is a bit much maybe (the default is 5, and probably good).

Seen this failure:
```
(line   41) broken    http://pacha.cafepais.com/ - HTTPConnectionPool(host='pacha.cafepais.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff9cc55a438>: Failed to establish a new connection: [Errno -2] Name or service not known'))
```

And I am surprised to see that it does not use an internal cache?!  (or does it just print the same result again)

```
writing output... [ 97%] usage

(line  707) redirect  http://bpaste.net - permanently to https://bpaste.net/
(line  367) redirect  http://docs.python.org/library/pdb.html - permanently to https://docs.python.org/3/library/pdb.html
(line  370) redirect  http://docs.python.org/library/pdb.html - permanently to https://docs.python.org/3/library/pdb.html
(line  332) redirect  http://docs.python.org/library/pdb.html - permanently to https://docs.python.org/3/library/pdb.html
(line  336) redirect  http://docs.python.org/library/pdb.html - permanently to https://docs.python.org/3/library/pdb.html
(line  336) redirect  http://docs.python.org/library/pdb.html - permanently to https://docs.python.org/3/library/pdb.html
```

Anyway, I suggest the following patch:

- given that it is a builtin module, the doc reference link is not really needed - sorry for not checking this before
- I think it should not be quiet, since it takes very long, and it is good to see progress

```diff
diff --git i/doc/en/conf.py w/doc/en/conf.py
index 93fbc4787..5c78bca38 100644
--- i/doc/en/conf.py
+++ w/doc/en/conf.py
@@ -112,16 +112,18 @@
 # A list of ignored prefixes for module index sorting.
 # modindex_common_prefix = []
 
-# A list of regular expressions that match URIs that should not be checked when doing a lin>
-# see https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-the-linkche>
+# A list of regular expressions that match URIs that should not be checked when
+# doing a linkcheck.
 linkcheck_ignore = [
     "https://github.com/numpy/numpy/blob/master/doc/release/1.16.0-notes.rst#new-deprecatio>
     "https://blogs.msdn.microsoft.com/bharry/2017/06/28/testing-in-a-cloud-delivery-cadence>
     "http://pythontesting.net/framework/pytest-introduction/",
+    r"https://github.com/pytest-dev/pytest/issues/\d+"
+    r"https://github.com/pytest-dev/pytest/pull/\d+"
 ]
 
-# The number of worker threads to use when checking links
-linkcheck_workers = 20
+# The number of worker threads to use when checking links (default=5).
+# linkcheck_workers = 5
 
 
 # -- Options for HTML output ---------------------------------------------------
diff --git i/tox.ini w/tox.ini
index 37577bdae..6a3c9b888 100644
--- i/tox.ini
+++ w/tox.ini
@@ -72,7 +72,7 @@ usedevelop = True
 changedir = doc/en
 deps = -r{toxinidir}/doc/en/requirements.txt
 commands =
-    sphinx-build -q -W --keep-going -b linkcheck . _build
+    sphinx-build -W --keep-going -b linkcheck . _build
 
 [testenv:doctesting]
 basepython = python3
```

Reply:
Let me know if I should just push it - I'd also squash it then already.

But there are still some 404s anyway - I've read somewhere that there's another PR/plan to fix them?

Reply:
Thanks for the suggestions.
Since you already spend the efforts I‘d be happy if you pushed this.
If it more convenient if I apply the patch, I’ll do this tonight

Reply:
Why ignore pytest GitHub links? Seems useful to know when you accidentally typo'd an issue/PR number and end up with an issue which doesn't exist.

Reply:
@The-Compiler 
Because there are a lot, and it slows the whole process down.
To check if they exist, you could basically just check if they are below the current maximum also (since issues redirect to pulls anyway).
Typos might happen, but it is usually for some other (existing) issue then.

Reply:
Still wonder about:

> Why is "https://github.com/numpy/numpy/blob/master/doc/release/1.16.0-notes.rst#new-deprecations" ignored?

Reply:
> Since you already spend the efforts I‘d be happy if you pushed this.

Can do so later (have it in a stash), if we all agree on it.
Apart from that I'd like to have it squashed into your commits anyway.

Reply:
I applied and pushed the patch.

> Why is "https://github.com/numpy/numpy/blob/master/doc/release/1.16.0-notes.rst#new-deprecations" ignored?

`linkcheck` treats this as _broken_, although it's working fine. Hence I added it to the list of false-positives.

Reply:
Personally, be removing `-q` it find the output very noise, but this might be a matter of personal taste.

Reply:
> And I am surprised to see that it does not use an internal cache?! (or does it just print the same result again)

`linkcheck` does not appear to do any caching here. Might be worth to propose an upstream PR later

Reply:
> Personally, be removing `-q` it find the output very noise, but this might be a matter of personal taste.

I got the impression that it would hang.. and since it runs for minutes.. but well.